sormas-ui/src/main/java/de/symeda/sormas/ui/campaign/campaigndata/CampaignDataView.java

```diff
@@ -50,21 +61,78 @@
 	private final CampaignDataGrid grid;
 	private CampaignFormDataFilterForm filterForm;
 
+	protected OptionGroup campaignFormElementImportance;
+
+	public static final String ONLY_IMPORTANT_FORM_ELEMENTS = "onlyImportantFormElements";
+
 	public CampaignDataView() {
 		super(VIEW_NAME);
 
 		criteria = ViewModelProviders.of(getClass()).get(CampaignFormDataCriteria.class);
 		grid = new CampaignDataGrid(criteria);
 
 		VerticalLayout mainLayout = new VerticalLayout();
-		mainLayout.addComponent(createFilterBar());
+		HorizontalLayout filtersLayout = new HorizontalLayout();
+
+		filtersLayout.setWidthFull();
+		filtersLayout.setMargin(false);
+		filtersLayout.setSpacing(true);
+
+		CampaignFormDataFilterForm filterBar = createFilterBar();
+		filtersLayout.addComponent(filterBar);
+		filtersLayout.setComponentAlignment(filterBar, Alignment.TOP_LEFT);
+		filtersLayout.setExpandRatio(filterBar, 0.8f);
+
+		OptionGroup importanceFilterLayout = createImportanceFilterSwitch();
+		filtersLayout.addComponent(importanceFilterLayout);
+		filtersLayout.setComponentAlignment(importanceFilterLayout, Alignment.TOP_RIGHT);
+		filtersLayout.setExpandRatio(importanceFilterLayout, 0.2f);
+
+		mainLayout.addComponent(filtersLayout);
+
+		filterForm.getField(CampaignFormDataCriteria.CAMPAIGN_FORM_META).addValueChangeListener(e -> {
+			Object value = e.getProperty().getValue();
+			if (value == null) {
+				campaignFormElementImportance.setEnabled(false);
```

Comment:
I had a look at it and think we should better hide the switch when no campaign form is selected.